test/sys/test_socket.rs

```diff
@@ -169,7 +169,7 @@ mod recvfrom {
 
     const MSG: &'static [u8] = b"Hello, World!";
 
-    fn sendrecv<F>(rsock: RawFd, ssock: RawFd, f: F) -> Option<SockAddr>
+    fn sendrecv<F>(rsock: RawFd, ssock: RawFd, max_recv_once: Option<usize>, f: F) -> Option<SockAddr>
```

Comment:
The `max_recv_once` argument seems specific to the GSO tests.  Could you try to make `sendrecv` remain more generic?  Perhaps you could make the new argument an optional `Fn` that would be called after each `recvfrom`.